picoscope/ps4000a.py

```diff
@@ -278,28 +283,68 @@ def _lowLevelGetTimebase(self, tb, noSamples, oversample, segmentIndex):
 
     def getTimeBaseNum(self, sampleTimeS):
         """Return sample time in seconds to timebase as int for API calls."""
-        maxSampleTime = (((2 ** 32 - 1) - 4) / 2e7)
 
-        if sampleTimeS <= 12.5E-9:
-            timebase = math.floor(math.log(sampleTimeS * 8E7, 2))
-            timebase = max(timebase, 0)
-        else:
-            # Otherwise in range 2^32-1
-            if sampleTimeS > maxSampleTime:
-                sampleTimeS = maxSampleTime
+        if self.model in ('4828'):
+            maxSampleTime = (((2 ** 32 - 1) + 1) / 8E7)
+
+            if sampleTimeS <= 12.5E-9:
+                timebase = 0
+            else:
+                # Otherwise in range 2^32-1
+                if sampleTimeS > maxSampleTime:
+                    sampleTimeS = maxSampleTime
+
+                timebase = math.floor((sampleTimeS * 2e7) + 1)
+
+        elif self.model in ('4444'):
+            maxSampleTime = (((2 ** 32 - 1) - 2) / 5.0E7)
+
+            if sampleTimeS <= 2.5E-9 and self.resolution == self.ADC_RESOLUTIONS["12"]:
+                timebase = 0
+            elif sampleTimeS <= 20E-9 and self.resolution == self.ADC_RESOLUTIONS["14"]:
+                timebase = 3
+            else:
+                # Otherwise in range 2^32-1
+                if sampleTimeS > maxSampleTime:
+                    sampleTimeS = maxSampleTime
 
-            timebase = math.floor((sampleTimeS * 2e7) + 1)
+                timebase = math.floor((sampleTimeS * 5.0E7) + 2)
+
+        else: # The original case from non "A" series
+            print("Note - the model you are using may not be fully supported")
```

Comment:
should be warning. Can you please use `stacklevel=2` https://docs.python.org/3.6/library/warnings.html#warnings.warn